### PR TITLE
107 implement transisthor for ressource entities

### PR DESCRIPTION
**Pull request by LucasTesnier**
Implement Entity transit operation (Cast / Uncast)

Use it inside Communicator.

Change RFC for Entity transfer.

Add unit and functional testing for this functionality.

### CODE REVIEW

**Review by 31Nathan — APPROVED** [review]
Need another review

**Review by 31Nathan — APPROVED** [review]
LGTM!

File: libs/Transisthor/Transisthor.hpp [hunk 1]
```diff
@@ -62,7 +68,7 @@ namespace transisthor_lib
         void *transitEcsDataToNetworkData(
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:68]: Add inline mention

File: libs/Transisthor/Transisthor.hpp [hunk 2]
```diff
@@ -72,16 +78,17 @@ namespace transisthor_lib
...
             }
             return networkObject;
         }
 
         /// @brief Function called by the ECS. The transfered data will be converted to a Network object and send to the
         /// communicator.
         /// @tparam C Type of the resource
-        /// @param id Id of the transfered resource
+        /// @param id Id of the
+        /// transferehttps://tekfeed.epitech.eu/#/universite/38/categorie/courses-for-epitech-studentsd resource
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:91]: I don’t think your college research should be here;)

File: libs/Transisthor/Transisthor.hpp [hunk 3]
```diff
@@ -95,19 +102,188 @@ namespace transisthor_lib
...
-        /// @param id Id of the transfered entity
-        /// @param type Type id of the transfered entity
-        /// @param ...args All components used inside the wanted entity.
-        template <std::derived_from<Component>... C>
-        void transitEcsDataToNetworkData(unsigned short id, unsigned short type, C &&...args)
+        /// @param id Id of the entity
+        /// @param type Type id of the entity
+        /// @param destination of the message
+        /// @return Return value his only used for testing (Unit and functional)
+        void *transitEcsDataToNetworkDataEntity(
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:109]: Add inline mention or transfert to a cpp file

File: libs/Transisthor/Transisthor.hpp [hunk 4]
```diff
@@ -95,19 +102,188 @@ namespace transisthor_lib
...
             /// WILL BE IMPLEMENTED WHEN ENTITY HAVE BEEN MERGED.
         }
 
+        /// @brief Function called by the ECS to transfer an order of creation for an AlliedProjectile entity
+        /// @param id Id of the new entity to create
+        /// @param posX X value for the Position component
+        /// @param posY Y value for the Position component
+        /// @param destination of the message
+        /// @return Return value his only used for testing (Unit and functional)
+        void *transitEcsDataToNetworkDataEntityAlliedProjectile(
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:125]: Add inline mention or transfert it into a cpp file

File: libs/Transisthor/Transisthor.hpp [hunk 5]
```diff
@@ -95,19 +102,188 @@ namespace transisthor_lib
...
+            return networkObject;
+        }
+
+        /// @brief Function called by the ECS to transfer an order of creation for an Enemy entity
+        /// @param id Id of the new entity to create
+        /// @param posX X value for the Position component
+        /// @param posY Y value for the Position component
+        /// @param destination of the message
+        /// @return Return value his only used for testing (Unit and functional)
+        void *transitEcsDataToNetworkDataEntityEnemy(
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:152]: Add inline mention or transfert it into a cpp file

File: libs/Transisthor/Transisthor.hpp [hunk 6]
```diff
@@ -95,19 +102,188 @@ namespace transisthor_lib
...
+            return networkObject;
+        }
+
+        /// @brief Function called by the ECS to transfer an order of creation for an EnemyProjectile entity
+        /// @param id Id of the new entity to create
+        /// @param posX X value for the Position component
+        /// @param posY Y value for the Position component
+        /// @param destination of the message
+        /// @return Return value his only used for testing (Unit and functional)
+        void *transitEcsDataToNetworkDataEntityEnemyProjectile(
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:179]: Add inline mention or transfert it into a cpp file

File: libs/Transisthor/Transisthor.hpp [hunk 7]
```diff
@@ -95,19 +102,188 @@ namespace transisthor_lib
...
+            return networkObject;
+        }
+
+        /// @brief Function called by the ECS to transfer an order of creation for an Obstacle entity
+        /// @param id Id of the new entity to create
+        /// @param posX X value for the Position component
+        /// @param posY Y value for the Position component
+        /// @param destination of the message
+        /// @return Return value his only used for testing (Unit and functional)
+        void *transitEcsDataToNetworkDataEntityObstacle(
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:206]: Add inline mention or transfert it into a cpp file

File: libs/Transisthor/Transisthor.hpp [hunk 8]
```diff
@@ -95,19 +102,188 @@ namespace transisthor_lib
...
+            return networkObject;
+        }
+
+        /// @brief Function called by the ECS to transfer an order of creation for an Player entity
+        /// @param id Id of the new entity to create
+        /// @param posX X value for the Position component
+        /// @param posY Y value for the Position component
+        /// @param destination of the message
+        /// @return Return value his only used for testing (Unit and functional)
+        void *transitEcsDataToNetworkDataEntityPlayer(
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:233]: Add inline mention or transfert it into a cpp file

File: libs/Transisthor/Transisthor.hpp [hunk 9]
```diff
@@ -95,19 +102,188 @@ namespace transisthor_lib
...
+
+        /// @brief Function called by the ECS to transfer an order of creation for an Projectile entity
+        /// @param id Id of the new entity to create
+        /// @param posX X value for the Position component
+        /// @param posY Y value for the Position component
+        /// @param velAbsc Abscissa value for the Velocity component
+        /// @param velOrd Ordonate value for the Velocity component
+        /// @param destination of the message
+        /// @return Return value his only used for testing (Unit and functional)
+        void *transitEcsDataToNetworkDataEntityProjectile(unsigned short id, int posX, int posY, double velAbsc,
```
Comment (ZQUEMA) [inline libs/Transisthor/Transisthor.hpp:262]: Add inline mention or transfert it into a cpp file

**Review by ZQUEMA — CHANGES_REQUESTED** [review]
Some changes are needed.

**Review by ZQUEMA — APPROVED** [review]
OK

**Review by TomDUVAL-MAHE — APPROVED** [review]
NTR

**Pull request merged**